### PR TITLE
For mozilla-mobile#27282: Remove WallpaperManager

### LINKED ISSUE #27282: Remove `WallpaperManager`
`WallpaperManager` is a leftover piece of tech debt from Wallpapers v1. It is time to remove it entirely. 

It looks like the only usage of the class is through `WallpaperManager.isDefaultTheCurrentWallpaper`, which can be moved and/or reworked into the companion of `Wallpaper`.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21722)

### PR DESCRIPTION
**Pull request by mbaimuratov**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #27282

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/Components.kt [hunk 1]
```diff
@@ -107,7 +93,9 @@ class Components(private val context: Context) {
 
     val addonCollectionProvider by lazyMonitored {
         // Check if we have a customized (overridden) AMO collection (supported in Nightly & Beta)
-        if (FeatureFlags.customExtensionCollectionFeature && context.settings().amoCollectionOverrideConfigured()) {
+        if (FeatureFlags.customExtensionCollectionFeature && context.settings()
+                .amoCollectionOverrideConfigured()
+        ) {
```
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/components/Components.kt:98]: It looks like there's a few unrelated changes that may have been auto-generated by a formatter. We try to keep our commits as small and focused as possible, so please revert these unrelated changes.

File: app/src/main/java/org/mozilla/fenix/components/Components.kt [hunk 2]
```diff
@@ -20,33 +19,20 @@ import mozilla.components.feature.addons.update.DefaultAddonUpdater
 import mozilla.components.feature.autofill.AutofillConfiguration
 import mozilla.components.lib.publicsuffixlist.PublicSuffixList
 import mozilla.components.support.base.worker.Frequency
-import org.mozilla.fenix.BuildConfig
-import org.mozilla.fenix.Config
-import org.mozilla.fenix.FeatureFlags
-import org.mozilla.fenix.HomeActivity
-import org.mozilla.fenix.R
+import org.mozilla.fenix.*
```
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/components/Components.kt:22]: These `*` imports will get caught by the linter during CI. Please revert them. You can run our lint checks locally with
`./gradlew ktlint`
and
`./gradlew detekt`

**Review by MatthewTighe — CHANGES_REQUESTED** [review]
This is looking pretty good so far! Looks like we just need to revert some changes to the `Components` file.

Also, we will want to land this patch as a single commit. It looks like there are 3 commits currently, so please remember to keep the branch squashed when it's ready for another round of review 🙂

**Review by MatthewTighe — APPROVED** [review]
This looks great, thanks for your contribution!

We have a [specific process](https://github.com/mozilla-mobile/fenix/wiki/Guide-to-merging-contributor-PRs) for merging contributor PRs. Basically, I have pulled down your commit, pushed it to my own fork, and opened https://github.com/mozilla-mobile/fenix/pull/27413 in order to merge it. You'll still get credit as the author of the commit though, if that's something you're interested in 🙂

**Comment by gabrielluong** [comment]
I am gonna close this in favor of https://github.com/mozilla-mobile/fenix/pull/27413 which will land your PR. Thank you for the patch!

**Pull request closed without merging**